sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorHandlerProvider.java

```diff
@@ -31,59 +29,75 @@
-     * @param transportType Transport type used for the connection.
-     * @param proxyOptions The options to use for proxy.
-     * @param product The name of the product this connection handler is created for.
-     * @param clientVersion The version of the client library creating the connection handler.
-     * @param clientOptions provided by user.
+     * @param options Options for the connection.
      * @return A new {@link ConnectionHandler}.
+     *
+     * @throws NullPointerException If {@code connectionId}, {@code productName}, {@code clientVersion},
+     *      {@code options} is {@code null}.
      */
-    public ConnectionHandler createConnectionHandler(String connectionId, String hostname,
-            AmqpTransportType transportType, ProxyOptions proxyOptions, String product, String clientVersion,
-            SslDomain.VerifyMode verifyMode, ClientOptions clientOptions) {
-        switch (transportType) {
+    public ConnectionHandler createConnectionHandler(String connectionId, String productName, String clientVersion,
+        ConnectionOptions options) {
+        Objects.requireNonNull(connectionId, "'connectionId' cannot be null.");
+        Objects.requireNonNull(options, "'options' cannot be null.");
+        Objects.requireNonNull(productName, "'productName' cannot be null.");
+        Objects.requireNonNull(clientVersion, "'clientVersion' cannot be null.");
+
+        switch (options.getTransportType()) {
             case AMQP:
-                return new ConnectionHandler(connectionId, hostname, product, clientVersion, verifyMode, clientOptions);
+                return new ConnectionHandler(connectionId, options.getHostname(), options.getPort(), productName,
+                    clientVersion, options.getSslVerifyMode(), options.getClientOptions());
             case AMQP_WEB_SOCKETS:
+                final ProxyOptions proxyOptions = options.getProxyOptions();
                 if (proxyOptions != null && proxyOptions.isProxyAddressConfigured()) {
-                    return new WebSocketsProxyConnectionHandler(connectionId, hostname, proxyOptions, product,
-                        clientVersion, verifyMode, clientOptions);
-                } else if (WebSocketsProxyConnectionHandler.shouldUseProxy(hostname)) {
-                    logger.info("System default proxy configured for hostname '{}'. Using proxy.", hostname);
-                    return new WebSocketsProxyConnectionHandler(connectionId, hostname,
-                        ProxyOptions.SYSTEM_DEFAULTS, product, clientVersion, verifyMode, clientOptions);
+                    // options.getPort() and options.getHostname() doesn't matter here because it will be overridden by
+                    // the proxy's port and hostname.
+                    return new WebSocketsProxyConnectionHandler(connectionId, options.getHostname(), options.getPort(),
+                        proxyOptions, productName, clientVersion, options.getSslVerifyMode(),
```

Comment:
If  `options.getPort() and options.getHostname() doesn't matter` : Should we have another constructor of `WebSocketsProxyConnectionHandler` which does not require these as a parameter. And we do not need to pass this ?